frontend/components/context/Notifications/NotificationProvider.tsx

```diff
@@ -0,0 +1,64 @@
+import { createContext, ReactNode, useContext, useState } from "react";
+
+import Notifications from "./Notifications";
+
+type NotificationType = "success" | "error";
+
+export type Notification = {
+  text: string;
+  type: NotificationType;
+};
+
+type NotificationContextState = {
+  createNotification: ({ text, type }: Notification) => void;
+};
+
+const NotificationContext = createContext<NotificationContextState>({
+  createNotification: () => console.log("createNotification not set!"),
+});
+
+export const useNotificationContext = () => useContext(NotificationContext);
```

Comment:
The `useNotificationContext` hook will allow all `NotificationContext` components to create notifications. Simply called this hook and destructure the `createNotification` function. It only returns `createNotification` for now, since I didn't see any obvious usage for `clearNotifications` or `notifications`